backend/app/resources/validate_utils.py

```diff
@@ -20,12 +24,56 @@ def validate_contact(contact, contact_str, error_list):
     return error_list
 
 
+def validate_role_info(role, role_info, role_info_str, error_list):
+    if not isinstance(role_info, dict) and role != USERINFO_ROLE_ADMIN:
+        error_list.append(f"The {role_info_str} supplied is not a dict.")
+        return error_list
+
+    asp_info_fields = ["num_kids"]
+    # donor_info_fields = ["type", "tags"]
+
+    if role == USERINFO_ROLE_ASP:
+        for field in asp_info_fields:
+            role_info = role_info["asp_info"]
+            if field not in role_info:
+                error_list.append(
+                    f'The {role_info_str} supplied does not have field "{field}".'
+                )
+            elif type(role_info[field]) is not int:
+                error_list.append(
+                    f'The field "{field}" in {role_info_str} is not a string.'
+                )
+            elif role_info[field] == "":
+                error_list.append(
+                    f'The field "{field}" in {role_info_str} must not be empty.'
+                )
+
+    # elif role == USERINFO_ROLE_DONOR:
+    #     for field in donor_info_fields:
+    #         role_info = role_info["donor_info"]
+    #         if field not in role_info:
+    #             error_list.append(
+    #                 f'The {role_info_str} supplied does not have field "{field}".'
+    #             )
+    #         elif type(role_info[field]) is not str:
+    #             error_list.append(f'The field "{field}" in {role_info_str}
+    #               is not a string.')
+    #         elif role_info[field] == "":
+    #             error_list.append(
+    #                 f'The field "{field}" in {role_info_str} must not be empty.'
+    #             )
```

Comment:
we should either leave a comment above this section explaining why this block of code is commented out and needs to be kept here. an explicit comment will provide valuable context for anyone else who needs to work with this file again.

if we can't explain why this needs to be here, then we should remove it.